sirbot/pythondev/slack.py

```diff
@@ -470,3 +472,13 @@ def add(self, slack):
                             " `{0}`".format(command.text)
 
         await slack.send(response)
+
+        async def members_joined(self, event, slack, _):
```

Comment:
you don't need to `await slack.channels` multiple time. It returns a channel object.

The most efficient would be to request it one time to check if the event is for #general and then a second time with the `update` argument to be sure we update the total members count. You can then use the object for the `to` and counting the number of members